### ISSUE TITLE
Some kind of loop when moving apps around

### ISSUE THREAD

**Issue by daPhipz** [stuarthayhurst/alphabetical-grid-extension#4]
I tested the code (`master`, commit 63bb153325357f02bbcc3ac8c702f07396b7eaaa) on Ubuntu 21.04 with GNOME 3.38:

The initial reordering after installing the extension works as expected (after restarting the shell, but that's a known issue), but when I reorder applications or move apps in or out of folders while the extension is active, the UI freezes for ~30sec. During that time, the fans of my laptop went crazy - the `gnome-shell` process took up to 100% CPU, I checked that via a second `tty` where I had `htop` running.

When the UI un-froze, I could see that `journalctl -f -o cat | grep -E 'alphabetical-app-grid'` (which I had running in the background to catch log messages) produced a ton of output - apparently, the extension was reordering the grid over and over. These log messages appeared in a loop when I tried to move an application to another spot in the grid:

```
alphabetical-app-grid: Running GNOME shell 3.38 or lower, skipping reload
alphabetical-app-grid: Reordered grid
```

**Comment by stuarthayhurst** [comment]
It appears the debouncing code that's supposed to prevent this isn't working... I'll create a VM for this and test tomorrow.

**Comment by stuarthayhurst** [comment]
From memory, this should be a pretty simple fix  I think I added some code that changes the order, before it works out whether the change it detected was external or not. This would end up creating a loop of: program makes change -> program detects change made by program -> repeat.

**Comment by daPhipz** [comment]
Ah, I see - unfortunately, my "programming" skills pretty much end with bash scripting, so I can't really help with that...

**Comment by stuarthayhurst** [comment]
If I'm right, It'll be like a 2 line fix, I'll get to it later

**Comment by stuarthayhurst** [comment]
Did `alphabetical-app-grid: Folders changed, triggering reorder` appear in the log?

**Comment by stuarthayhurst** [comment]
Didn't mean to close it without checking the fix works for you, GitHub scanned the commit message :/
Anyway, I managed to reproduce the issue and that commit should fix it :)

**Comment by daPhipz** [comment]
> Did `alphabetical-app-grid: Folders changed, triggering reorder` appear in the log?

I will check that later today, and verify if the fix works for me.

Maybe you know it already (in which case sorry for explaining it 🙈), but you can re-open issues if you want ;)

**Comment by stuarthayhurst** [comment]
I left it closed because I could reproduce it and fixed it, but I'll reopen the issue if it isn't fixed for you.

**Comment by daPhipz** [comment]
> Did `alphabetical-app-grid: Folders changed, triggering reorder` appear in the log?

Nope, it did not.

The commit did fix it, I tested `master` at 75d547d98febca773178ce19c1d91328aebaf0f7 and everything works as expected. Thank you for the quick fix!

**Comment by stuarthayhurst** [comment]
Perfect, thanks for picking up on this before release :)
